src/DssSpell.sol

```diff
@@ -18,52 +18,262 @@ pragma solidity 0.6.12;
+}
+
+interface Hopeable {
+    function hope(address) external;
+}
+
+interface Kissable {
+    function kiss(address) external;
+}
+
+interface RwaLiquidationLike {
+    function ilks(bytes32) external returns (string memory,address,uint48,uint48);
+    function init(bytes32, uint256, string calldata, uint48) external;
 }
 
 contract DssSpellAction is DssAction {
 
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
-    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/6d2eaab1360395916b9b09c4137c5da6f2de9f1a/governance/votes/Executive%20vote%20-%20July%2017%2C%202021.md -q -O - 2> /dev/null)"
-    string public constant override description =
-        "2021-07-17 MakerDAO Executive Spell | Hash: 0x887d27e20f47d0701d1eea14045bb5fdbea634401f7dc5aedd729af13a5c8ddc";
+    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/2a7a8c915695b7298fe725ee3dc6c613fa9d9bbe/governance/votes/Executive%20vote%20-%20April%2012%2C%202021.md -q -O - 2>/dev/null)"
+    string public constant override description = "RWA003-RWA006 integration";
 
-    // Turn off office hours
-    function officeHours() public override returns (bool) {
-        return false;
-    }
+    // Many of the settings that change weekly rely on the rate accumulator
+    // described at https://docs.makerdao.com/smart-contract-modules/rates-module
+    // To check this yourself, use the following rate calculation (example 8%):
+    //
+    // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
+    //
+    // A table of rates can be found at
+    //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
+    //
+    uint256 constant TWO_PCT            = 1000000000627937192491029810;
+    uint256 constant FOUR_PT_FIVE_PCT   = 1000000001395766281313196627;
+    uint256 constant SIX_PCT            = 1000000001847694957439350562;
+    uint256 constant SEVEN_PCT          = 1000000002145441671308778766;
 
-    uint256 constant WAD     = 10 ** 18;
-    uint256 constant MILLION = 10 ** 6;
+    // Math
+    uint256 constant THOUSAND = 10 ** 3;
+    uint256 constant MILLION  = 10 ** 6;
+    uint256 constant WAD      = 10 ** 18;
+    uint256 constant RAY      = 10 ** 27;
+    uint256 constant RAD      = 10 ** 45;
 
-    address constant MIP21_LIQUIDATION_ORACLE = 0x88f88Bb9E66241B73B84f3A6E197FbBa487b1E30;
+    // Maker changelog 
+    address public constant MAKER_CHANGELOG = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
 
     function actions() public override {
+        // https://vote.makerdao.com/polling/QmRCn7Mh#poll-detail
+        CentrifugeCollateralValues memory RWA003 = CentrifugeCollateralValues({
+            MCD_JOIN: 0x1Fe789BBac5b141bdD795A3Bc5E12Af29dDB4b86,
+            GEM: 0x07F0A80aD7AeB7BfB7f139EA71B3C8f7E17156B9,
+            OPERATOR: 0x2A9798c6F165B6D60Cfb923Fe5BFD6f338695D9B,
```

Comment:
`liq` is unset on this contract--will that be taken care of by your post-spell integration tasks?

Reply:
Exactly, we'll do that in the integration spell we'll run on our side after the executive vote.